src/docfx/build/xref/XrefMap.cs

```diff
@@ -3,46 +3,24 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
-using System.Threading;
 
 namespace Microsoft.Docs.Build
 {
     internal class XrefMap
     {
-        // TODO: key could be uid+moniker+locale
         private readonly IReadOnlyDictionary<string, Lazy<ExternalXrefSpec>> _externalXrefMap;
         private readonly IReadOnlyDictionary<string, InternalXrefSpec> _internalXrefMap;
 
-        private static ThreadLocal<Stack<(string uid, string propertyName, Document parent)>> t_recursionDetector = new ThreadLocal<Stack<(string, string, Document)>>(() => new Stack<(string, string, Document)>());
-
         public XrefMap(Context context, Docset docset, RestoreFileMap restoreFileMap)
         {
             _internalXrefMap = InternalXrefMapBuilder.Build(context);
             _externalXrefMap = ExternalXrefMapLoader.Load(docset, restoreFileMap);
         }
 
-        public (Error error, string href, string display, IXrefSpec xrefSpec) Resolve(string uid, SourceInfo<string> href, string displayPropertyName, string text, Document relativeTo)
+        public IXrefSpec Resolve(SourceInfo<string> uid)
         {
-            if (t_recursionDetector.Value.Contains((uid, displayPropertyName, relativeTo)))
```

Comment:
>             [](start = 0, length = 12)

It can be removed totally as we talked offline, but we can simplify the stack, only keep the uid